### ISSUE TITLE
Fix Illegal Alignments in CmpLog

### ISSUE THREAD

**Issue by domenukk** [AFLplusplus/AFLplusplus#2171]
The current `CmpLog` implementation accesses memory in an unaligned way in both, the instrumentation, and afl-fuzz. This is UB and works on x86 and aarch64(depending on configuration), but it will trap on other architectures.

Specifically, the cmplog struct should be padded to be a multiple of 64 bit:
https://github.com/AFLplusplus/AFLplusplus/blob/stable/include/cmplog.h#L53-L63

The (slower, but potentially more mem-efficient) option is to access all entries of this struct byte by byte, instead of reading 64 bits at a time.

**Comment by tokatoka** [comment]
why do we have these unused fields?
```
  u64 unused;
  u8  unused1;
  u8  unused2;
```

**Comment by vanhauser-thc** [comment]
So this would satisfy everything right?

```
--- a/include/cmplog.h
+++ b/include/cmplog.h
@@ -56,18 +56,17 @@ struct cmp_operands {
 struct cmp_operands {
 
   u64 v0;
-  u64 v1;
   u64 v0_128;
+  u64 v0_256_0;  // u256 is unsupported by any compiler for now, so future use
+  u64 v0_256_1;
+  u64 v1;
   u64 v1_128;
-  u64 unused;
-  u8  unused1;
-  u8  unused2;
+  u64 v1_256_0;
+  u64 v1_256_1;
+   u8 unused[8];
 
 } __attribute__((packed));
 
 struct cmpfn_operands {
 
   u8 v0[32];
-  u8 v0_len;
   u8 v1[32];
+  u8 v0_len;
   u8 v1_len;
+  u8 unused[6];
 
 } __attribute__((packed));
```

=> `sizeof(cmpfn)=72, sizeof(cmp)=72`

**Comment by domenukk** [comment]
Fixed in 6e37f9b
